src/lib.rs

```diff
@@ -156,10 +157,290 @@ impl Color for RGBA {
+pub struct HSL {
+    // hue
+    pub h: Angle,
+
+    // saturation
+    pub s: u8,
+
+    // luminosity
+    pub l: u8,
+}
+
+impl fmt::Display for HSL {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "hsl({}, {}%, {}%)", self.h.degrees(), self.s, self.l)
+    }
+}
+
+impl HSL {
+    /// Transforms numerical values into a HSL struct.
+    ///
+    /// # Example
+    /// ```
+    /// use {css_colors::HSL, css_colors::Angle};
+    ///
+    /// let salmon = HSL::new(Angle::new(6), 93, 71);
+    ///
+    /// assert_eq!(salmon, HSL { h: Angle { degrees: 6 }, s: 93, l: 71 });
+    /// ```
+    pub fn new(h: Angle, s: u8, l: u8) -> HSL {
+        HSL { h, s, l }
+    }
+}
+
+impl Color for HSL {
+    fn to_css(self) -> String {
+        self.to_string()
+    }
+
+    fn to_rgb(self) -> RGB {
+        // FIXME: create impl, add tests for this
+        RGB::new(0, 0, 0)
+    }
+
+    fn to_rgba(self) -> RGBA {
+        // FIXME: create impl, add tests for this
+        RGBA::new(0, 0, 0, 0)
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+/// A struct to represent how much hue, saturation, and luminosity should be added to create a color.
+/// Also handles alpha specifications.
+///
+/// A valid value for `h` must range between `0-360`.
+/// The saturation ranges between `0-100`, where `0` is completely desaturated, and `100` is full saturation.
+/// The luminosity ranges between `0-100`, where `0` is no light (black), and `100` is full light (white).
+///
+/// For more, see the [CSS Color Spec](https://www.w3.org/TR/2018/REC-css-color-3-20180619/#hsla-color).
+pub struct HSLA {
+    // hue
+    pub h: Angle,
+
+    // saturation
+    pub s: u8,
+
+    // luminosity
+    pub l: u8,
+
+    // alpha
+    pub a: u8,
+}
+
+impl fmt::Display for HSLA {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "hsla({}, {}%, {}%, {:.02})", self.h, self.s, self.l, self.a as f32 / 255.0)
+    }
+}
+
+impl HSLA {
+    /// Transforms numerical values into a HSL struct.
+    ///
+    /// # Example
+    /// ```
+    /// use {css_colors::HSLA, css_colors::Angle};
+    /// let light_salmon = HSLA::new(Angle::new(6), 93, 71, 128);
+    ///
+    /// assert_eq!(light_salmon, HSLA { h: Angle { degrees: 6 }, s: 93, l: 71, a: 128 });
+    /// ```
+    pub fn new(h: Angle, s: u8, l: u8, a: u8) -> HSLA {
+        HSLA { h, s, l, a }
+    }
+}
+
+impl Color for HSLA {
+    fn to_css(self) -> String {
+        self.to_string()
+    }
+
+    fn to_rgb(self) -> RGB {
+        // FIXME: create impl, add tests for this
+        RGB::new(0, 0, 0)
+    }
+
+    fn to_rgba(self) -> RGBA {
+        // FIXME: create impl, add tests for this
+        RGBA::new(0, 0, 0, 0)
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+/// A struct that represents the number of degrees in a circle.
+/// Legal values range from `0-359`. Anything else is unsused.
+pub struct Angle {
+    pub degrees: u16,
```

Comment:
this should not be public, otherwise someone could construct an invalid angle without going through `new` (or assign an invalid value after construction)